internal/pubsub/samples/pub/rabbit.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/streadway/amqp"
+	"gocloud.dev/internal/pubsub"
+	"gocloud.dev/internal/pubsub/rabbitpubsub"
+)
+
+const rabbitURL = "amqp://guest:guest@localhost:5672/"
+
+// openRabbitTopic returns the RabbitMQ topic for the given topic ID.
+func openRabbitTopic(topicID string) (*pubsub.Topic, func(), error) {
+	conn, err := amqp.Dial(rabbitURL)
```

Comment:
It is surprising that this doesn't take a context. What happens if the server isn't running?

Reply:
It quickly returns with an error:
```
[ ~/gocloud.dev/internal/pubsub/samples/pub ] pub -rabbiturl=amqp://nonexistent -env rabbit foo 
pub: dial tcp: lookup nonexistent: no such host
[ ~/gocloud.dev/internal/pubsub/samples/pub ] pub -rabbiturl=amqp://localhost:8080 -env rabbit foo 
pub: dial tcp 127.0.0.1:8080: connect: connection refused
```